server/apps/ldap/tests.py

```diff
@@ -7,26 +7,94 @@
-            auth_user = get_resource_service('auth').authenticate(user)
-            self.assertEquals(auth_user['username'], user['username'])
+            auth_user = get_resource_service('auth').authenticate({'username': 'barf', 'password': 'dummy'})
+            self.assertEqual(auth_user['username'], user['user_to_import'])
 
-            cmd.run(user['username'], user['password'], user['user_to_import'])
-            auth_user2 = get_resource_service('auth').authenticate(user)
-            self.assertEquals(auth_user2['username'], user['username'])
+            cmd.run(user['username'], user['password'], 'BARF')
+            auth_user2 = get_resource_service('auth').authenticate({'username': 'barf', 'password': 'dummy'})
+            self.assertEqual(auth_user2['username'], user['user_to_import'])
+
+            self.assertEqual(auth_user2['_id'], auth_user['_id'])
+
+        def test_import_profile_for_already_imported_user_raises_exception(self, mock_ldap_connection):
+            with mock.patch('apps.ldap.ldap.get_user', return_value={'username': 'user'}):
+                with self.assertRaises(SuperdeskApiError) as context:
+                    service = get_resource_service('import_profile')
+                    doc = {'username': 'user', 'password': 'pwd', 'profile_to_import': 'barf'}
+                    users = service.post([doc])
+                    self.assertIsNotNone(users)
+                    self.assertEqual(len(users), 1)
+                    doc = {'username': 'user', 'password': 'pwd', 'profile_to_import': 'BARF '}
+                    service.post([doc])
+
+                ex = context.exception
+                self.assertEqual(ex.message, 'User already exists in the system.')
+                self.assertEqual(ex.status_code, 400)
+                self.assertDictEqual(ex.payload, {'profile_to_import': 1})
+
+        def test_import_profile_by_user_other_than_logged_in_user_raises_exception(self, mock_ldap_connection):
+            with mock.patch('apps.ldap.ldap.get_user', return_value={'username': 'user1'}):
+                with self.assertRaises(SuperdeskApiError) as context:
+                    service = get_resource_service('import_profile')
+                    doc = {'username': 'user', 'password': 'pwd', 'profile_to_import': 'barf'}
+                    service.post([doc])
+
+                ex = context.exception
+                self.assertEqual(ex.message, 'Invalid Credentials.')
+                self.assertEqual(ex.status_code, 403)
+                self.assertDictEqual(ex.payload, {'credentials': 1})
+
+        def test_user_import_profile_with_invalid_credentials(self, mock_ldap_connection):
+            with mock.patch('apps.ldap.ldap.ADAuth.authenticate_and_fetch_profile',
+                            side_effect=CredentialsAuthError(credentials={'username': 'test'}, error='test')):
+                with mock.patch('apps.ldap.ldap.get_user', return_value={'username': 'user'}):
```

Comment:
To avoid too much indentation, it might be better to patch the test method itself, e.g.:

```py
@mock.patch('apps.ldap.ldap.ADAuth.authenticate_and_fetch_profile', side_effect=...)
@mock.patch('apps.ldap.ldap.get_user', return_value={'username': 'user'})
def test_user_import_profile_with_invalid_credentials(self, mock_ldap_connection):
    # ... method body
```